src/bundles/ipfs-desktop.js

```
@@ -8,6 +10,10 @@ if (window.ipfsDesktop) {
   bundle = {
     ...bundle,
     reducer: (state = {}, action) => {
+      if (action.type === ACTIONS.EXP_TOGGLE) {
+        window.ipfsDesktop.toggleSetting(`experiments.${action.payload.key}`)
```

I would suggest you restrict this toggle action to only execute on known desktop experiments, otherwise if we add webui only ones in the future it will also trigger this method.

---

we could pass around the `desktop: bool` in the action if that would be the easiest?

---

It would not be a problem, because there wouldn't be any handler for that event then. But I made a change to fix it: window.ipfsDesktop now has an array with experiments.

---

In that case perhaps it's better to fail silently rather than having an additional experiments array to update?

---

Yeah, I'll just revert my update then :)

---

Done